docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,246 @@ If you need to have external labels in your query results, this is currently not
+   
+   ```bash
+   # Get list of all blocks in the bucket
+   thanos tools bucket inspect \
+       --objstore.config-file bucket.yaml \
+       --output=tsv > blocks.tsv
+       
+   # Find blocks from replica that we will drop    
+   cat blocks.tsv| grep prometheus_replica=<PROMETHEUS-REPLICA-TO-DROP> \
+       | awk '{print $1}' > blocks_to_drop.tsv
+       
+   # Mark found blocks for deletion
+   for ID in $(cat blocks_to_drop.tsv)
+   do
+       thanos tools bucket mark \
+          --marker="deletion-mark.json" \
+          --objstore.config-file bucket.yaml \
+          --details="Removed as duplicate" \
+          --id $ID
+   done
+   ```
+   
+   > **Note:** Replace `prometheus_replica` with the unique label that would differentiate Prometheus replicas in your setup.
+
+   Clean up the duplicate blocks marked for deletion again:
+  
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+   > **Tip:** If you want to visualize exactly what is happening in the blocks, with respect to the source of blocks, external labels, compaction levels, and more, you can use the following command to get the output as CSV and import it into a Google spreadsheet:
+   > ```bash 
+   > thanos tools bucket inspect \
+   >     --objstore.config-file bucket-prod.yaml \
+   >     --output=csv > thanos-blocks.csv
+   > ```
+
+4. Relabel the blocks with external labels.
+
+   Mimir doesn’t inject external labels from the `meta.json` file into query results. Therefore, you need to relabel the blocks with the required external labels in the `meta.json` file. 
+   
+   > **Note:** You can get the external labels in the `meta.json` file of each block from the CSV file that is imported (as mentioned in the preceding tip), and build the rewrite configuration accordingly.
+   
+   Create a rewrite configuration that is similar to this:
+   
+   ```bash
+   # relabel-config.yaml
+   - action: replace
+     target_label: "<LABEL-KEY>"
+     replacement: "<LABEL-VALUE"
+   ```
+   
+   Perform the rewrite dry run to confirm all works well.
+   
+   ```bash
+   # Get list of all blocks in the bucket after removing the depuplicate and downsampled blocks.
+   thanos tools bucket inspect \
+       --objstore.config-file bucket.yaml \
+       --output=tsv > blocks-to-rewrite.tsv
+   
+   # Check if rewrite of the blocks with external labels is working as expected.
+   for ID in $(cat blocks-to-rewrite.tsv)
+   do
+       thanos tools bucket rewrite \
+           --objstore.config-file bucket.yaml \
+           --rewrite.to-relabel-config-file relabel-config.yaml \
+           --dry-run \
+           --id $ID
+   done
+   ```
+   
+   After you confirm that the rewrite is working as expected via `--dry-run`, apply the changes with the `--no-dry-run` flag. Remember to include `--delete-blocks, otherwise the original blocks will not be marked for deletion.
```

Comment:
```suggestion
   After you confirm that the rewrite is working as expected via `--dry-run`, apply the changes with the `--no-dry-run` flag. Remember to include `--delete-blocks`, otherwise the original blocks will not be marked for deletion.
```